### ISSUE TITLE
darknet_face_with_landmark

### ISSUE THREAD

**Issue by ouyanghuiyu** [AlexeyAB/darknet#5651]
I modified some files for face detection (with landmark): https://github.com/ouyanghuiyu/darknet_face_with_landmark
![image](https://user-images.githubusercontent.com/42023607/82172892-1f273600-98fe-11ea-85a4-9f8b9680c1c0.png)

**Comment by AlexeyAB** [comment]
@ouyanghuiyu 

* Did you use this dataset? http://shuoyang1213.me/WIDERFACE/index.html

* Did you check accuracy by using codalab? https://competitions.codalab.org/competitions/20146

* Did you just add 5 (x,y) coordinates for landmarks? 
* What activation did you use for them? 
* Did you make them relative to the image size or to the object bbox (since all landmarks are always inside the object bbox)?

* Did you try to use yolov4?

**Comment by ouyanghuiyu** [comment]
I trained with widerface：http://shuoyang1213.me/WIDERFACE/index.html
I check accuracy with widerface val dateset
i add 5 (x,y) coordinates for landmarks
i don't use activation funtion,by linear
i use distance, use wing loss
i trained it with mobilenetv2 + ssh,don't use yolov4:https://github.com/ouyanghuiyu/darknet_face_with_landmark/blob/master/cfg/mbv2_yolov3_face.cfg 
https://github.com/ouyanghuiyu/darknet_face_with_landmark/blob/master/models/mbv2_yolov3_face_final.weights
i trained a light model now

**Comment by WongKinYiu** [comment]
i modified [image.c.txt](https://github.com/AlexeyAB/darknet/files/4648321/image.c.txt) and [detector.c.txt](https://github.com/AlexeyAB/darknet/files/4648327/detector.c.txt).
now you can use `./darknet detector test ...` to get prediction.
![predictions](https://user-images.githubusercontent.com/12152972/82288762-cb881b80-99d5-11ea-9519-2ad7f6018c92.jpg)

**Comment by ouyanghuiyu** [comment]
@WongKinYiu  It's great,but the image.c is gone，I can't download it !

**Comment by WongKinYiu** [comment]
@ouyanghuiyu link is fixed. https://github.com/AlexeyAB/darknet/issues/5651#issuecomment-630591883

**Comment by ouyanghuiyu** [comment]
@WongKinYiu It's right!

**Comment by ouyanghuiyu** [comment]
@WongKinYiu Hello, may I ask how to control not to draw the label

**Comment by WongKinYiu** [comment]
@ouyanghuiyu 

i temporally comment line 631 of image.c.
`                draw_label(im, top + width, left, label, rgb);`
to
`                //draw_label(im, top + width, left, label, rgb);`

**Comment by ouyanghuiyu** [comment]
@WongKinYiu ok,thanks!

**Comment by blackwool** [comment]
> 
> 
> @ouyanghuiyu
> 
> i modified [image.c](https://github.com/AlexeyAB/darknet/files/4648321/image.c.txt) and [detector.c](https://github.com/AlexeyAB/darknet/files/4648327/detector.c.txt).
> now you can use `./darknet detector test ...` to get prediction.
> ![predictions](https://user-images.githubusercontent.com/12152972/82288762-cb881b80-99d5-11ea-9519-2ad7f6018c92.jpg)

@WongKinYiu
Hi ,the result of face detect is great,but where are the cfg and weights file for face and landmark detection ?
Thanks a lot !

**Comment by WongKinYiu** [comment]
@blackwool 

in @blackwool 's repo.
https://github.com/ouyanghuiyu/darknet_face_with_landmark

**Comment by nikkymen** [comment]
@WongKinYiu

What input network size did you use to get this result on selfie.jpg?

**Comment by WongKinYiu** [comment]
@nikkymen 

i use default size.
https://github.com/ouyanghuiyu/darknet_face_with_landmark/commit/a4fcefb404cbdb6f4106c3ff2ce0d231152ecacd#diff-ef44c622e08357e8eb8e51620cc221c7L5-L6

**Comment by AlexeyAB** [comment]
@WongKinYiu Did you use 416x416 to get detection with such small objects?

**Comment by WongKinYiu** [comment]
@AlexeyAB No, the original default size is 2048x1280.

**Comment by nikkymen** [comment]
@WongKinYiu 
This is really a good result, congrats!

**Comment by Chappie74** [comment]
> @AlexeyAB No, the original default size is 2048x1280.

is this referring to your training images size or the size in the .cfg file? 
Also, very good results. It amazed me how it was able to detect even those small faces in the back.

**Comment by xjohnxjohn** [comment]
@ouyanghuiyu You said "i trained it with mobilenetv2 + ssh", what the parameter in cfg that you use it?

**Comment by ouyanghuiyu** [comment]
@xjohnxjohn mobilenetv2 + ssh is about 2.489  BFLOPS when input 416*416 , also,you can  trian a more light model with my repo https://github.com/ouyanghuiyu/darknet_face_with_landmark

**Comment by qustions** [comment]
@ouyanghuiyu Good work how much fast and accurate is this compare to retinaface if possible can you write size and speed benchmark in your repo.

**Comment by zpmmehrdad** [comment]
@AlexeyAB Hi,

Can we use it in the latest version?

**Comment by AlexeyAB** [comment]
@zpmmehrdad No, it is implemented only in this fork: https://github.com/ouyanghuiyu/darknet_face_with_landmark

**Comment by blackwool** [comment]
> 
> 
> @xjohnxjohn mobilenetv2 + ssh is about 2.489 BFLOPS when input 416*416 , also,you can trian a more light model with my repo https://github.com/ouyanghuiyu/darknet_face_with_landmark

I test this model( 2.489 BFLOPS)  to detector face, in win10,GTX1060, it needs about 113ms, but when i test yolov3 model ( 65.9 BFlops), It  takes about 37ms  . I do not know why the model is so slow but with very small BFLOPS , Do you test it for inferencing?

**Comment by blackwool** [comment]
I printed out milli-seconds cost for each layer of this network ,and find that  convolution layers with large groups will cost a lot of time . like layers 52,56,60 etc. So ,even if it has small BFLOPS, it also runs slower than yolov3
